python/ray/train/batch_predictor.py

```diff
@@ -421,15 +421,12 @@ def _determine_preprocessor_batch_format(
         dataset_block_format = ds.dataset_format()
         if dataset_block_format == BlockFormat.SIMPLE:
             # Naive case that we cast to pandas for compatibility.
+            # TODO: Revisit
             return BatchFormat.PANDAS
 
         if not preprocessor:
-            # No preprocessor, just use the dataset format.
-            return (
-                BatchFormat.NUMPY
-                if dataset_block_format == BlockFormat.ARROW
-                else BatchFormat.PANDAS
-            )
+            # No preprocessor, just use the predictor format.
+            return self._predictor_cls._batch_format_to_use()
```

Comment:
this function should never be called in the first place if `preprocessor` is None. Don't think we need this if clause

Reply:
@amogkam Hmm seems like this is still called if `preprocessor is None`? https://github.com/ray-project/ray/blob/dbc3bd89bec364c66041f8de48afc772cb19054c/python/ray/train/batch_predictor.py#L188

Reply:
yeah, but it doesn't need to be. But this is a minor point, so looks good to merge.